java/device/iothub-java-client/src/main/java/com/microsoft/azure/iothub/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -346,5 +348,24 @@ private void updateConnectionOptions(String userName, String userPassword)
         this.connectionOptions.setMqttVersion(mqttVersion);
         this.connectionOptions.setUserName(userName);
         this.connectionOptions.setPassword(userPassword.toCharArray());
+        this.setEnabledProtocal(enabledProtocols);
+    }
```

Comment:
connectionOptions is a MqttConnectOptions member in org.eclipse.paho.client.mqttv3.MqttConnectOptions and it is a third-party package, so I can not simply modfiy it.